bindings/python/translator.py

```diff
@@ -0,0 +1,109 @@
+    """
+    num_workers: int
+    cache: int
+    logging: str
+    terminology: str
+    force_terminology: bool
+
+    _config: bergamot.ServiceConfig
+    _model: bergamot.TranslationModel
+    _responseOpts: bergamot.ResponseOptions
+    _service: bergamot.Service
+
+    def __init__(self, model_conifg_path: str, num_workers: int=1, cache: int=0, \
+                 logging="off", terminology: str="", force_terminology: bool=False):
+        """Initialises the translator class
+
+        :param model_conifg_path: Path to the configuration file for the translation model.
+        :param num_workers: Number of CPU workers.
+        :param cache: cache size. 0 means no cache.
+        :param logging: Log level: trace, debug, info, warn, err(or), critical, off.
+        :param terminology: Path to terminology file, TSV format
+        :param force_terminology: Force terminology to appear on the target side. May impact translation quality.
+        """
+        self.num_workers = num_workers
+        self.cache = cache
+        self.logging = logging
+        self.terminology = terminology
+        self.force_terminology = force_terminology
+
+        self._config = bergamot.ServiceConfig(self.num_workers, self.cache, self.logging, self.terminology, self.force_terminology)
+        self._service = bergamot.Service(self._config)
+        self._responseOpts = bergamot.ResponseOptions() # Default false for all, if we want to enable HTML later, from here
+        self._model = self._service.modelFromConfigPath(model_conifg_path)
+
+    def resetTerminology(self, terminology: str="", force_terminology: bool=False) -> None:
+        """Resets the terminology of the model
+        :param terminology: path to the terminology file.
+        :param force_terminology: force terminology
+        :return: None
+        """
+        self.terminology = terminology
+        self.force_terminology = force_terminology
+        self._config = bergamot.ServiceConfig(self.num_workers, self.cache, self.logging, self.terminology, self.force_terminology)
+        self._service = bergamot.Service(self._config)
+
+    def resetNumWorkers(self, num_workers) -> None:
+        """Resets the number of workers
+        :param num_workers: number of parallel CPU threads.
+        :return: None
+        """
+        self.num_workers = num_workers
+        self._config = bergamot.ServiceConfig(self.num_workers, self.cache, self.logging, self.terminology, self.force_terminology)
+        self._service = bergamot.Service(self._config)
+    
+    def translate(self, sentences: List[str]) -> str:
+        """Translates a list of strings
+        :param sentences: A List of strings to be translated.
+        :return: Translation output.
+        """
+        responses = self._service.translate(self._model, bergamot.VectorString(sentences), self._responseOpts)
+        ret = ""
+        for response in responses:
+            ret = ret + response.target.text
+        return ret
```

Comment:
Why require a list of inputs, but always only return a single string as output? I'd say do either one or the other, and I don't see a reason why we can't return a list in the output, e.g. `return [response.target.text for response in responses]`. Also saves you making assumptions on how to concatenate the output. There might be a need for spaces there.